File: google_oauth/gen_auth_url.js
 const clientId = '905804377606-2hfg5p2dsht9o528ckrt0r27bc20cn66.apps.googleusercontent.com';
 const redirectUri = 'https://fe-1n-test.trustdice.win/';
 const GOOGLE_REDIRECT_URI = redirectUri; // Or replace with your actual value
 const scope = 'https://www.googleapis.com/auth/userinfo.email';
 const state = 'wozhaoritianzuiniubi';

 const rawGetParams = {
     response_type: 'code',
     client_id: clientId, // Replace `clientId` with your actual client ID
     redirect_uri: GOOGLE_REDIRECT_URI,
     scope: scope,
     state: state,
     access_type: 'offline',
     include_granted_scopes: 'true',
     // login_hint: 'hint@example.com',
     prompt: 'consent',
 };

 // Function to encode parameters as a query string
 const urlencode = (params) => {
     return Object.keys(params)
         .map(key => `${encodeURIComponent(key)}=${encodeURIComponent(params[key])}`)
         .join('&');
 };

 const encodedParams = urlencode(rawGetParams);
 const authorizationUrl = 'https://accounts.google.com/o/oauth2/auth?' + encodedParams;

 console.log(authorizationUrl);

File: google_oauth/gen_auth_url.py
from urllib.parse import urlencode
import json


file_path = 'client_secret.json'

# Open the file and load its contents
with open(file_path, "r") as json_file:
    data = json.load(json_file)
    print('data', data)
    client_id = data['web']['client_id']
    client_secret = data['web']['client_secret']

# redirect_uri = GOOGLE_REDIRECT_URI
redirect_uri = 'https://www.baidu.com'
scope = 'https://www.googleapis.com/auth/userinfo.email'

# state: 自定义的字段， 生成 authorization url 的时候确定
#  后续拿到 auth code 的时候这个 state 也会返回
state = 'wozhaoritianzuiniubi'

raw_get_params = {
    'response_type': 'code',
    'client_id': client_id,
    'redirect_uri': redirect_uri,
    'scope': scope,
    'state': state,
    'access_type': 'offline',
    'include_granted_scopes': 'true',
    # 'login_hint': 'hint@example.com',
    'prompt': 'consent',
}

encoded_params = urlencode(raw_get_params)
authorization_url = 'https://accounts.google.com/o/oauth2/auth?' + encoded_params

authorization_url

